pkg/etcdutil/etcdutil.go

```diff
@@ -294,3 +321,88 @@ func TakeAndSaveFullSnapshot(ctx context.Context, client client.MaintenanceClose
 
 	return snapshot, nil
 }
+
+// checkFullSnapshotIntegrity verifies the integrity of the full snapshot by comparing
+// the appended SHA256 hash of the full snapshot with the calculated SHA256 hash of the full snapshot data.
+func checkFullSnapshotIntegrity(snapshotData io.ReadCloser, snapTempDBFilePath string, logger *logrus.Entry) (io.ReadCloser, error) {
```

Comment:
Can you add Unit tests

Reply:
It will be difficult to write a unit test for this function:
1. It's taking data stream as input which comes from full snapshot data of snapshot api call which was called on etcd.
2. Actually this function is already getting tested when integration tests triggers a full snapshot which internally calls this function to verify full snapshot integrity.

Reply:
I see your point about integration tests, but they might not cover all scenarios for the checkFullSnapshotIntegrity function. 

We could use a dummy snapshot in unit tests to check these scenarios more thoroughly.